driver/src/main/scala/api/commands/AtlasSearchAggregation.scala

```diff
@@ -1115,23 +1125,61 @@ private[commands] trait AtlasSearchAggregation[P <: SerializationPack] {
        * import reactivemongo.api.bson.collection.BSONCollection
        *
        * def prepare(coll: BSONCollection): coll.AggregationFramework.AtlasSearch.Near = coll.AggregationFramework.AtlasSearch.Near(
-       *   origin = Near.origin(1),
+       *   origin = Near.numericOrigin(1),
        *   path = "field",
        *   pivot = Some(0.5D)
        * )
        * }}}
        */
+      import java.time.Instant
+
       def apply(
         origin: Near.Origin,
         path: SearchString,
         pivot: Option[Double] = None,
         score: Option[Score] = None): Near =
         new Near(origin, path, pivot, score)
 
-      final class Origin private[api] (val value: pack.Value)
+      private[api] trait Origin {
+        def value: pack.Value
+      }
+
+      private[api] final class InstantOrigin(origin: Instant) extends Origin {
```

Comment:
There is no need for those subclasses.

The conversion can be directly done in the date & numeric factories, then having the private BSON representation is sufficient.